src/context/Locale.js

```diff
@@ -18,6 +19,12 @@ import {
 } from "../languages";
 import { languageSelector } from "../reducers/settings";
 
+if ("__setDefaultTimeZone" in Intl.DateTimeFormat) {
+  /** https://formatjs.io/docs/polyfills/intl-datetimeformat/#default-timezone */
+  // $FlowFixMe
+  Intl.DateTimeFormat.__setDefaultTimeZone(getTimeZone()); // eslint-disable-line no-underscore-dangle
+}
+
```

Comment:
Fixing the DateTimeFormat timezone issue: 
  - **`src/context/Locale.js`**:
    Using `react-native-localize`'s `getTimeZone()` and using it as a default timezone for `DateTimeFormat`.